app/data/models/application.js

```diff
@@ -1,9 +1,25 @@
-module.exports = (sequelize, DataTypes) => sequelize.define('application', {
-  applicationId: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
-  reference: DataTypes.STRING
-},
-{
-  tableName: 'application',
-  freezeTableName: true,
-  timestamps: false
-})
+module.exports = (sequelize, DataTypes) => {
+  return sequelize.define('application', {
+    id: {
+      type: DataTypes.BIGINT,
+      primaryKey: true,
+      autoIncrement: true
+    },
+    reference: {
+      type: DataTypes.STRING,
+      primaryKey: true,
+      set (val) {
+        this.setDataValue('reference', val.toUpperCase())
+      }
+    },
+    type: { type: DataTypes.STRING, primaryKey: true, allowNull: false },
```

Comment:
Why is `type` set as a primary key - I am genuinely curious as opposed to saying it isn't correct.

Also, is it possible to set `type` as a enum and state what values are valid? I'm not familiar enough to know if it is possible.